src/main/java/dev/j3fftw/litexpansion/machine/WaterSprinkler.java

```diff
@@ -97,47 +90,53 @@ public int getRadius() {
     }
 
     @Override
-    protected void tick(Block b) {
-        BlockMenu inv = BlockStorage.getInventory(b);
+    protected void tick(@Nonnull Block b) {
+        final BlockMenu inv = BlockStorage.getInventory(b);
 
         if (b.getRelative(BlockFace.DOWN).getType() == Material.WATER) {
-            inv.replaceExistingItem(PROGRESS_SLOT, waterFoundItem);
+            if (inv.hasViewer()) {
+                inv.replaceExistingItem(PROGRESS_SLOT, waterFoundItem);
+            }
         } else {
-            inv.replaceExistingItem(PROGRESS_SLOT, noWaterItem);
+            if (inv.hasViewer()) {
```

Comment:
Just do it at the top if...

Reply:
Need the return if it doesnt have water

Reply:
Put it in a variable, move it above. If no water then return, if (viewers) { if (water) { } else { } }

Reply:
The return cancels everything after it too, I cant have it only execute if a viewer is present